app/src/main/java/woowacourse/shopping/productlist/ProductListViewModel.kt

```diff
@@ -0,0 +1,49 @@
+package woowacourse.shopping.productlist
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import woowacourse.shopping.ShoppingRepository
+import woowacourse.shopping.domain.Product
+import woowacourse.shopping.uimodel.ProductUiModel
+import woowacourse.shopping.uimodel.toProductUiModel
+
+class ProductListViewModel(
+    private val repository: ShoppingRepository,
+) : ViewModel() {
+    private val _products: MutableLiveData<List<Product>> = MutableLiveData()
+    val products: LiveData<List<Product>> get() = _products
+
+    private val _productUiModels: MutableLiveData<List<ProductUiModel>> = MutableLiveData()
+    val productUiModels: LiveData<List<ProductUiModel>> get() = _productUiModels
+
+    private val _currentSize: MutableLiveData<Int> = MutableLiveData()
+    val currentSize: LiveData<Int> get() = _currentSize
+
+    val totalSize: Int = repository.productsTotalSize()
+
+    init {
+        loadProducts(PRODUCTS_START_POSITION)
+    }
+
+    fun loadProducts(startPosition: Int) {
+        runCatching {
+            repository.products(startPosition, PRODUCTS_OFFSET_SIZE)
+        }.onSuccess { loadedProducts ->
+            val currentItems = products.value ?: emptyList()
+            val currentUiItems = productUiModels.value ?: emptyList()
```

Comment:
orEmpty() 확장 함수를 활용해보세요!
```suggestion
            val currentItems = products.value?.orEmpty()
            val currentUiItems = productUiModels.value?.orEmpty()
```

Reply:
`?:` 연산자를 사용하지 않고 해당 확장 함수를 사용하는 방법도 있었군요!
감사합니다! 참고해서 활용해보도록 하겠습니다~~ 😄